webhook_server.py

```
@@ -0,0 +1,245 @@
+    signature = request.headers.get("X-Hub-Signature-256")
+    
+    # Verify signature
+    if not await verify_webhook_signature(raw_body, signature):
+        raise HTTPException(status_code=401, detail="Invalid signature")
+    
+    # Parse the JSON payload
+    try:
+        payload_data = json.loads(raw_body.decode('utf-8'))
+        payload = GitHubWebhookPayload(**payload_data)
+    except (json.JSONDecodeError, ValueError) as e:
+        logger.error(f"Invalid JSON payload: {e}")
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)